src/Microsoft.DotNet.ImageBuilder/src/Commands/GenerateEolAnnotationDataOptions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Linq;
+using static Microsoft.DotNet.ImageBuilder.Commands.CliHelper;
+
+#nullable enable
+namespace Microsoft.DotNet.ImageBuilder.Commands;
+
+public class GenerateEolAnnotationDataOptions : Options
+{
+    public string EolDigestsListPath { get; set; } = string.Empty;
+    public string OldImageInfoPath { get; set; } = string.Empty;
+    public string NewImageInfoPath { get; set; } = string.Empty;
+    public bool AnnotateEolProducts { get; set; }
+    public string RepoPrefix { get; set; } = string.Empty;
+    public string RegistryName { get; set; } = string.Empty;
+}
+
+public class GenerateEolAnnotationDataOptionsBuilder : CliOptionsBuilder
+{
+    public override IEnumerable<Option> GetCliOptions() =>
+        base.GetCliOptions()
+            .Concat(
+                [
+                    CreateOption<bool>("annotate-eol-products", nameof(GenerateEolAnnotationDataOptions.AnnotateEolProducts),
+                        "Annotate images of EOL products"),
+                ]
+            );
+
+    public override IEnumerable<Argument> GetCliArguments() =>
+        base.GetCliArguments()
+            .Concat(
+                [
+                    new Argument<string>(nameof(AnnotateEolDigestsOptions.EolDigestsListOutputPath),
```

Comment:
```suggestion
                    new Argument<string>(nameof(GenerateEolAnnotationDataOptions.EolDigestsListPath),
```

I think this one is a bug - it causes an exception later on when you run the code.